### PR TITLE
InfluxDB: Improve handling of intervals in alerting

### PR DESCRIPTION
**Pull request by gabor**
InfluxDB in InfluxQL mode: we can simply use the interval-value that comes in `DataQuery.Interval`, and send it to the database. no need to do any manual calculation.

(this was enabled by https://github.com/grafana/grafana/pull/38434 merged)

NOTE: the code only handles the first query in the query-array. there was a check before, to return an error when the query-array-length is zero. i also return an error if the query-array-length is more than 1

### CODE REVIEW

File: pkg/tsdb/influxdb/query.go [hunk 1]
```diff
@@ -29,16 +29,13 @@ func (query *Query) Build(queryContext *backend.QueryDataRequest) (string, error
...
+	intervalText := interval.FormatDuration(query.Interval)
+	intervalMs := int64(query.Interval / time.Millisecond)
 
 	res = strings.ReplaceAll(res, "$timeFilter", query.renderTimeFilter(queryContext))
-	res = strings.ReplaceAll(res, "$interval", i.Text)
-	res = strings.ReplaceAll(res, "$__interval_ms", strconv.FormatInt(i.Milliseconds(), 10))
-	res = strings.ReplaceAll(res, "$__interval", i.Text)
+	res = strings.ReplaceAll(res, "$interval", intervalText)
+	res = strings.ReplaceAll(res, "$__interval_ms", strconv.FormatInt(intervalMs, 10))
+	res = strings.ReplaceAll(res, "$__interval", intervalText)
```
Comment (dsotirakis) [inline pkg/tsdb/influxdb/query.go:38]: Why are we switching back to use `interval.Calculator` instead of `tsdb.Calculator`?

Comment (gabor) [reply pkg/tsdb/influxdb/query.go:38]: sorry, that was not the plan, i'll fix it 👍

Comment (gabor) [reply pkg/tsdb/influxdb/query.go]: done.

File: pkg/tsdb/influxdb/model_parser.go [hunk 2]
```diff
@@ -40,11 +47,30 @@ func (qp *InfluxdbQueryParser) Parse(model *simplejson.Json, dsInfo *models.Data
...
-		return nil, err
+	interval := query.Interval
+
+	// in old-alert use-case query.Interval is zero.
+	// for this case, to be backawrd-compatible,
+	// we calculate the final alert based on:
+	// - time-range
+	// - min-interval in json
+	// - min-interval from data-source-config
+	if interval == 0*time.Nanosecond {
```
Comment (marefr) [inline pkg/tsdb/influxdb/model_parser.go:58]: Can we update old alert to populate query.Interval properly and get rid of this extra code in "every" data source?

Comment (gabor) [reply pkg/tsdb/influxdb/model_parser.go:58]: you're right, i created a POC at https://github.com/grafana/grafana/pull/38434 , i can simplify this PR afterwards.

Comment (gabor) [reply pkg/tsdb/influxdb/model_parser.go]: that PR is merged now, i will update this PR.

**Review by marefr — APPROVED** [review]
LGTM code wise. Expect you to have tested it

**Pull request merged**